src/backend/distributed/commands/multi_copy.c

```diff
@@ -610,7 +839,16 @@ CopyToNewShards(CopyStmt *copyStatement, char *completionTag)
 			{
 				SendCopyBinaryFooters(copyOutState, shardConnections->connectionList);
 				FinalizeCopyToNewShard(shardConnections);
-				UpdateShardStatistics(relationId, shardConnections->shardId);
+
+				if (masterConnection == NULL)
+				{
+					UpdateShardStatistics(relationId, shardConnections->shardId);
+				}
+				else
+				{
+					UpdateRemoteShardStatistics(masterConnection, relationName,
```

Comment:
It would be kind be nice if we computed shardlength, shardminvalue, shardmaxvalue during the COPY so we could avoid this 3-way interaction and instead just send an UPDATE to the master. The benefit of the current approach is that it doesn't require much code, but I think computing it on-the-fly would be an improvement.